sdk/storage/azure-storage-blob/tests/test_helpers_async.py

```diff
@@ -60,3 +65,103 @@ async def read(self, size: int = -1) -> bytes:
         self._offset += len(data)
 
         return data
+
+
+class MockAioHttpClientResponse(ClientResponse):
+    def __init__(
+        self, url: str,
+        body_bytes: bytes,
+        headers: Optional[Dict[str, Any]] = None,
+        status: int = 200,
+        reason: str = "OK"
+    ) -> None:
+        super(MockAioHttpClientResponse).__init__()
+        self._url = url
+        self._body = body_bytes
+        self._headers = headers
+        self._cache = {}
+        self._loop = None
+        self.status = status
+        self.reason = reason
+
+
+class MockStorageTransport(AsyncHttpTransport):
+    async def send(self, request: HttpRequest, **kwargs: Any) -> RestAioHttpTransportResponse:
+        if request.method == 'GET':
+            # download_blob
+            rest_response = RestAioHttpTransportResponse(
+                request=request,
+                internal_response=MockAioHttpClientResponse(
+                    request.url,
+                    b"Hello Async World!",
+                    {
+                        "Content-Type": "application/octet-stream",
+                        "Content-Range": "bytes 0-27/28",
+                        "Content-Length": "28",
```

Comment:
Please make `Content-Range` and `Content-Length` match the actual length of the content.